pkg/client/patch.go

```diff
@@ -112,6 +112,8 @@ func (s *mergeFromPatch) Data(obj Object) ([]byte, error) {
 
 		modified = modified.DeepCopyObject().(Object)
 		modified.SetResourceVersion(version)
+	} else if modified.GetResourceVersion() == "" {
```

Comment:
This isn't the fakeClient. This is the real client.  Are we supposed to be modifying this rather than `pkg/client/fake/client.go`?

Reply:
I believe you want to look around here 
https://github.com/kubernetes-sigs/controller-runtime/blob/d0396a3d6f9fb554ef2da382a3d0bf05f7565e65/pkg/client/fake/client.go#L426

Reply:
Thanks for the review. I'll look into it

Reply:
Hi Troy! I've fixed it using the "debug stack hack".

Reply:
Thx!